core/src/components/input/input.tsx

```diff
@@ -293,6 +301,7 @@ export class Input implements ComponentInterface {
   }
 
   componentDidLoad() {
+    this.originalIonInput = this.ionInput;
```

Comment:
Came across this when working on the `ion-range` v7 work, should this be in `connectedCallback` instead of waiting for the component to load?

Reply:
What were you running into on range?

Reply:
I did not run into a specific issue, but was viewing the change top-down in `ion-range` and thinking about the potential test cases that exist.

Waiting until after `render` for storing a reference to the original event, seems strange. If for nothing other than logical organization, I recommend we move this behavior up into `connectedCallback` so that it has no risk of the event being re-assigned before we are able to cache it:

https://stenciljs.com/docs/component-lifecycle

Reply:
I played around with this some. The practical difference is that `connectedCallback` fires again if the component is removed and re-added to the DOM, and it turns out this can cause problems if the component's `debounce` is defined on reconnect. This causes `originalIonInput` to be set to a debounced version of the event emitter. So if `debounce` is later set to `undefined`, the event stays debounced. Screencast: https://user-images.githubusercontent.com/90629384/195137814-4215a298-4e37-47a7-8148-3c07c185ac4e.mp4

This is obviously an extremely specific set of circumstances, but it may be indicative of other similar issues, so I'd rather keep the `originalIonInput` init where it is since the benefit of moving it seems to be mainly semantic. (You may want to adjust `ion-range` though 👀)

Reply:
Sounds good, thanks for doing the extra discovery here!